src/renderer/components/badge/badge.tsx

```diff
@@ -40,12 +40,26 @@ export const Badge = withTooltip(observer(({
   children,
   ...elemProps
 }: BadgeProps) => {
-  const elem = createRef<HTMLDivElement>();
+  const elem = useRef<HTMLDivElement>(null);
+  const elemHasBeenSet = useRef(false);
   const [isExpanded, setIsExpanded] = useState(false);
+  const [isExpandable, setIsExpandable] = useState(expandable);
 
-  const isExpandable = expandable && elem.current
-    ? elem.current.clientWidth < elem.current.scrollWidth
-    : false;
+  useEffect(() => {
+    if (elemHasBeenSet.current) {
+      return;
+    }
+
+    if (elem.current) {
+      elemHasBeenSet.current = true;
+    }
+
+    setIsExpandable((
```

Comment:
This has to be state because we need to trigger a re-render if it is now `true` (to add the class names)